components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2611,8 +2677,19 @@ where
                 self.fsm.peer.tag, self.fsm.peer.pending_merge_state, target, result
             );
         }
-        // If the merge succeed, all source peers is impossible in apply snapshot state
+        // Because of the checking before proposing `PrepareMerge`, which is
+        // no compact proposal between the smallest commit index and the latest index.
```

Comment:
The term `compact proposal` is quite strange.

Reply:
I change to `CompactLog` proposal.